packages/neuron-ui/src/widgets/DropdownWithCustomRender/index.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { ReactNode, useEffect, useMemo, useRef, useState } from 'react'
+import { LineDownArrow } from 'widgets/Icons/icon'
+
+import styles from './index.module.scss'
+
+export interface OptionProps {
+  label: string | ReactNode
+  value: string
+  className?: string
+  data?: Record<string, string | number>
+  index?: number
+}
+
+interface DropdownWithCustomRenderProps {
+  onChange: (changeValue: OptionProps) => void
+  options: OptionProps[]
+  value?: string
+  placeholder?: string
+  onFocus?: (isOpen: boolean) => void
+  disabled?: boolean
+  className?: string
+}
+
+const DEFAULT_PLACEHOLDER_STRING = 'Please select...'
+
+const DropdownWithCustomRender = ({
+  options: optionsFromProps,
+  onChange,
+  value: valueFromProps = '',
+  placeholder,
+  onFocus,
+  disabled,
+  className,
+}: DropdownWithCustomRenderProps) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const [isMounted, setIsMounted] = useState<boolean>(true)
```

Comment:
initial value of `isMounted` should be `false` according to https://github.com/nervosnetwork/neuron/pull/2599/files#diff-c69edb9c0ce9ae6d9eb94be1cf1f617a43cfe450ecf1dffc51dd5cce636cc1c3R167

```
  useEffect(() => {
    document.addEventListener('click', handleDocumentClick, false)
    document.addEventListener('touchend', handleDocumentClick, false)
    setIsMounted(true)

    return () => {
      setIsMounted(false)
      document.removeEventListener('click', handleDocumentClick, false)
      document.removeEventListener('touchend', handleDocumentClick, false)
    }
  }, [])
```